src/CoreWCF.Primitives/src/CoreWCF/Channels/TextMessageEncoderFactory.cs

```diff
@@ -395,18 +396,25 @@ public override bool IsContentTypeSupported(string contentType)
                 return false;
             }
 
-            public override Message ReadMessage(ArraySegment<byte> buffer, BufferManager bufferManager, string contentType)
+            public override ValueTask<Message> ReadMessageAsync(ReadOnlySequence<byte> buffer, BufferManager bufferManager, string contentType)
             {
                 if (bufferManager == null)
                 {
-                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(bufferManager));
+                    throw Fx.Exception.ArgumentNull(nameof(bufferManager));
                 }
 
                 Message message;
 
                 UTF8BufferedMessageData messageData = TakeBufferedReader();
                 messageData.Encoding = GetEncodingFromContentType(contentType, _contentEncodingMap);
-                messageData.Open(buffer, bufferManager);
+
+                int bufferLength = (int)buffer.Length;
+                byte[] bytes = bufferManager.TakeBuffer(bufferLength);
```

Comment:
UTF8BufferedMessageData should handle ReadOnlySequence natively. If the ReadOnlySequence can be wrapped as a stream (and I think I saw some code elsewhere where you do that via a Sequence), the TakeReader which calls XmlDictionaryReader.CreateTextReader can call the overload of CreateTextReader which takes a stream.

Reply:
@mconnew fixed in https://github.com/CoreWCF/CoreWCF/pull/801/commits/5106720621414b5d49047297eec45cc8a30aec1e